source/includes/inkling-reference/_schemas.html.md

```diff
@@ -1,265 +1,480 @@
-  | 
-  '(' <fieldDeclarationList> ')'  # anonymous reference
+```inkling--code
+  schema MNIST_schema
+    String label,           # primitive type
+    Luminance(28, 28) image # structured type
+  end
+
+  schema my_schema
+    Int32 x,                # primitive type
+    Int32{1:5} z            # primitive type with type constraint
+  end
 ```
 
-Select the Syntax tab to view the schema reference syntax.
+To the right you will see a few example schemas. 
+The last field `z` in `my_schema` has constrained type `Int32{1:5}`. For more information on this feature see [Constrained Types][1].
 
-A named schema is referenced by its name. An anonymous schema is referenced by its list of fields.
 
-## Schema Example
 
-```inkling--code
-schema MNIST_training_data_schema
-  UInt8 label,
-  Luminance(28, 28) image
-end
+## Schema References
+
+> Schema Reference Syntax
+
+```inkling--syntax
+schemaRef :=                 
+     '('   ')'                  # empty anonymous schema
+ |   '(' <name> ')'             # named schema
+ |   '('  <fieldDclnList> ')    # non-empty anonymous schema
 ```
 
-Select the Inkling tab to show a schema that has a field with a primitive type and a field with a structured type.
+Inkling statements can reference schemas by name. In addition, anywhere a schema name can
+be referenced, a list of fields can appear. This is an anonymous schema. 
+Anonymous schema references can also be empty, if allowed in context.
+
+## Types
+
+Inkling supports both primitive types and structured types.
 
-## Inkling Types
 
 ###### Primitive Types
 
+> Primitive Types List
+
 ```inkling--syntax
 primitiveType ::=
   Double | Float64 | Float32 | Int8 | Int16 | Int32 |
   Int64 | UInt8 | UInt16 | UInt32  | UInt64 | Bool | String
 ```
 
-Select the Syntax tab to show the Inkling set of primitive types which are used in schema declarations. The
-integer suffix indicates the size in bits of the type. Integer types beginning
-with 'U' are unsigned. 
+The Inkling set of primitive types includes numeric, string, and boolean types.
+
+To the right you will see the set of primitive types which are used in schema declarations. 
+
+The integer suffix indicates the size in bits of the type. 
+
+Integer types beginning with 'U' are unsigned. 
+
+`Double` and `Float64` are synonyms.
 
 ###### Structured Types
 
+> Structured Types List
+
 ```inkling--syntax
-structure_type ::= 
-  Luminance | Matrix
+structure_type ::=                      # syntax
+  Luminance 
+  | Matrix
 ```
 
-[//]: # (Are structured types used for simulators ever?)
+```inkling--code
+schema BallLocationSchema               # example
+  Luminance(84, 336) pixels,    
+  Matrix(UInt32, 1, 2) location 
+end
+```
 
+Structured types in Inkling are intended to support common
+Machine Learning types. The Machine Learning types currently supported include
+`Luminance` and `Matrix`.
+  
 Inkling currently supports the types Matrix and Luminance. (This list will be
```

Comment:
This sentence is redundant with the previous sentence.

Reply:
Fixed.